src/coreclr/nativeaot/Runtime/gctoclreventsink.cpp

```diff
@@ -11,17 +11,21 @@ void GCToCLREventSink::FireDynamicEvent(const char* eventName, void* payload, ui
 {
     LIMITED_METHOD_CONTRACT;
 
-#ifndef FEATURE_NATIVEAOT
     const size_t EventNameMaxSize = 255;
 
     WCHAR wideEventName[EventNameMaxSize];
-    if (MultiByteToWideChar(CP_ACP, 0, eventName, -1, wideEventName, EventNameMaxSize) == 0)
+    int i = 0;
+    while (true)
     {
-        return;
+        wideEventName[i] = (WCHAR)eventName[i];
```

Comment:
You may want to check that it is not overflowing the `EventNameMaxSize` buffer size.